src/PassPlanFilePlugin.groovy

```diff
@@ -0,0 +1,71 @@
+import static TerraformEnvironmentStage.PLAN
+import static TerraformEnvironmentStage.APPLY
+
+class PassPlanFilePlugin implements TerraformPlanCommandPlugin, TerraformApplyCommandPlugin, TerraformEnvironmentStagePlugin {
+
+    public static void init() {
+        PassPlanFilePlugin plugin = new PassPlanFilePlugin()
+
+        TerraformEnvironmentStage.addPlugin(plugin)
+        TerraformPlanCommand.addPlugin(plugin)
+        TerraformApplyCommand.addPlugin(plugin)
+    }
+
+    @Override
+    public void apply(TerraformEnvironmentStage stage) {
+        stage.decorate(PLAN,  archivePlanFile(stage.getEnvironment()))
+        stage.decorate(APPLY, downloadArchive(stage.getEnvironment()))
+    }
+
+    @Override
+    public void apply(TerraformPlanCommand command) {
+        String env = command.getEnvironment()
+        command.withArgument("-out=tfplan-" + env)
+    }
+
+    @Override
+    public void apply(TerraformApplyCommand command) {
+        String env = command.getEnvironment()
+        command.withDirectory("tfplan-" + env)
+    }
+
+    public Closure archivePlanFile(String env) {
+        return { closure ->
+            closure()
+            echo "Archiving tfplan-${env} file"
+            archiveArtifacts artifacts: "tfplan-" + env
+        }
+    }
+
+    public Closure downloadArchive(String env) {
+        return { closure ->
+            String jenkinsUrl  = Jenkinsfile.instance.getEnv()['JENKINS_URL']
+            String jobName     = Jenkinsfile.instance.getEnv()['JOB_NAME']
+            String buildNumber = Jenkinsfile.instance.getEnv()['BUILD_NUMBER']
+            String url = getArtifactUrl(jenkinsUrl, jobName, buildNumber, env)
+            echo "Downloading tfplan-${env} file from ${url}"
+
+            sh "wget -O tfplan-${env} ${url}"
+
```

Comment:
Same as above - is there a reason to use archive vs stash?  The above is  creating a dependency on `wget`.